server/src/test/java/io/crate/auth/HostBasedAuthenticationTest.java

```diff
@@ -82,13 +85,13 @@ public class HostBasedAuthenticationTest extends ESTestCase {
     private static final Settings HBA_2 = Settings.builder()
         .put("auth.host_based.config.2.user", "crate")
         .put("auth.host_based.config.2.address", "0.0.0.0/0")
-        .put("auth.host_based.config.2.method", "fake")
+        .put("auth.host_based.config.2.method", "password")
         .put("auth.host_based.config.2.protocol", "pg")
         .build();
 
     private static final Settings HBA_3 = Settings.builder()
         .put("auth.host_based.config.3.address", "127.0.0.1")
-        .put("auth.host_based.config.3.method", "md5")
+        .put("auth.host_based.config.3.method", "password")
         .put("auth.host_based.config.3.protocol", "pg")
         .build();
```

Comment:
We didn't take client method/payload to account and just silently returned NULL before. 
Tests where this was used are not about method  anyway: one is about CIDR, another is about settings to map conversion, no need to use invalid values.